plugins.go

```diff
@@ -106,6 +107,27 @@ func (p *Plugins) Build() (err error) {
 	return
 }
 
+// BuildAsync will build all of the plugins asynchronously
+func (p *Plugins) BuildAsync(q *queue.Queue) (err error) {
+	p.mu.Lock()
+	defer p.mu.Unlock()
+
+	var wg sync.WaitGroup
+	wg.Add(len(p.ps))
+
+	var errs errors.ErrorList
+	for _, pi := range p.ps {
+		q.New(func() {
```

Comment:
Does `q.New` call the anonymous function in a go-routine?

Reply:
`q.New` pushes the func to the queue :)

Reply:
If it is in a queue, what makes it asynchronous? Does the queue trigger builds in different goroutines?

Reply:
the queue has x number of goroutines listening to it. So it allows you to easily spread tasks over x threads. Check out the queue library :) https://github.com/hatchify/queue